packages/x-date-pickers/src/internals/hooks/useField/useField.ts

```diff
@@ -423,7 +424,13 @@ export const useField = <
     onChange: handleInputChange,
     onKeyDown: handleInputKeyDown,
     onMouseUp: handleInputMouseUp,
-    error: inputError,
     ref: handleRef,
   };
+
+  return {
+    ...response,
+    // only override when `error` is undefined.
+    // in case of multi input fields, the `error` value is provided externally and will always be defined
+    error: response.error ?? inputError,
```

Comment:
Only override the `error` flag to internal value if it is still undefined

Reply:
Another way of doing it is to return `otherForwartedProps.error` inside the `inputError` `useMemo` and avoid all this `response` variable.
Unless I'm missing something

It also shows that we should add `error` to `UseFieldForwardedProps` like we are adding the callbacks.


```diff
diff --git a/packages/x-date-pickers/src/internals/hooks/useField/useField.ts b/packages/x-date-pickers/src/internals/hooks/useField/useField.ts
index 16d7f8618..1902e8fa7 100644
--- a/packages/x-date-pickers/src/internals/hooks/useField/useField.ts
+++ b/packages/x-date-pickers/src/internals/hooks/useField/useField.ts
@@ -63,6 +63,7 @@ export const useField = <
       onBlur,
       onMouseUp,
       onPaste,
+      error,
       ...otherForwardedProps
     },
     fieldValueManager,
@@ -369,10 +370,15 @@ export const useField = <
     valueManager.defaultErrorState,
   );
 
-  const inputError = React.useMemo(
-    () => fieldValueManager.hasError(validationError),
-    [fieldValueManager, validationError],
-  );
+  const inputError = React.useMemo(() => {
+    // only override when `error` is undefined.
+    // in case of multi input fields, the `error` value is provided externally and will always be defined
+    if (error) {
+      return error;
+    }
+
+    return fieldValueManager.hasError(validationError);
+  }, [fieldValueManager, validationError, error]);
 
   React.useEffect(() => {
     // Select the right section when focused on mount (`autoFocus = true` on the input)
@@ -411,8 +417,8 @@ export const useField = <
     return 'tel';
   }, [selectedSectionIndexes, state.sections]);
 
-  const response = {
-    error: undefined,
+  return {
+    error: inputError,
     ...otherForwardedProps,
     value: valueStr,
     inputMode,
@@ -426,11 +432,4 @@ export const useField = <
     onMouseUp: handleInputMouseUp,
     ref: handleRef,
   };
-
-  return {
-    ...response,
-    // only override when `error` is undefined.
-    // in case of multi input fields, the `error` value is provided externally and will always be defined
-    error: response.error ?? inputError,
-  };
 };
diff --git a/packages/x-date-pickers/src/internals/hooks/useField/useField.types.ts b/packages/x-date-pickers/src/internals/hooks/useField/useField.types.ts
index ed0a9405e..980d17d68 100644
--- a/packages/x-date-pickers/src/internals/hooks/useField/useField.types.ts
+++ b/packages/x-date-pickers/src/internals/hooks/useField/useField.types.ts
@@ -85,6 +85,7 @@ export interface UseFieldForwardedProps {
   onClick?: () => void;
   onFocus?: () => void;
   onBlur?: () => void;
+  error?: boolean;
 }
 
 export type UseFieldResponse<TForwardedProps extends UseFieldForwardedProps> = Omit<
```

Reply:
Thanks for a cleaner suggestion. Did apply it with a minor fix. 😉